src/base/http/connection.cpp

```diff
@@ -33,29 +34,121 @@
 #include <QTcpSocket>
 
 #include "base/logger.h"
-#include "irequesthandler.h"
 #include "requestparser.h"
-#include "responsegenerator.h"
 
 using namespace Http;
 
-Connection::Connection(QTcpSocket *socket, IRequestHandler *requestHandler, QObject *parent)
-    : QObject(parent)
-    , m_socket(socket)
-    , m_requestHandler(requestHandler)
+Connection::Connection(QTcpSocket *socket, QObject *parent)
+    : QObject {parent}
+    , m_socket {socket}
 {
     m_socket->setParent(this);
-    m_idleTimer.start();
-    connect(m_socket, &QTcpSocket::readyRead, this, &Connection::read);
+
+    connect(m_socket, &QAbstractSocket::readyRead, this, &Connection::read);
+    connect(m_socket, &QAbstractSocket::bytesWritten, this, &Connection::bytesWritten);
+    connect(m_socket, &QAbstractSocket::disconnected, this, &Connection::disconnected);
+}
+
+void Connection::sendStatus(const Http::ResponseStatus &status)
+{
+    Q_ASSERT(m_status == PendingStatus);
+    m_socket->write(QString("HTTP/%1 %2 %3")
+                    .arg("1.1", // TODO: depends on request
+                         QString::number(status.code), status.text)
+                    .toLatin1()
+                    .append(Http::CRLF));
+    m_status = PendingHeaders;
+}
+
+void Connection::sendHeaders(const Http::HeaderMap &headers)
+{
+    Q_ASSERT(m_status == PendingHeaders);
+
+    const auto writeHeader = [this](const QString &key, const QString &value)
```

Comment:
`[m_socket]`

Reply:
I don't think you can capture member variable like this.